### PR TITLE
Update zfs-snapshot.8 to fix a small inaccuracy in the description of snapshot atomicity

### PR DESCRIPTION
**Pull request by shawnbayern**
Fixes a small inaccuracy in the description of snapshot atomicity

zfs-snapshot(8) appears to contain a small error.  The existing version reads "Snapshots are taken atomically, so that all snapshots correspond to the same moment in time."  Per zfs_main.c, which in do_snapshot() simply loops over argv, this does not appear to be correct when multiple snapshots are specified explicitly on the command line.  I believe the intent of the man page was to say that *recursive* snapshots are all created atomically.

This proposed change fixes that error.  Because the existing statement may confuse some readers anyway, the commit also also adds a small amount of general explanatory information that may be helpful.

The change also adds an introductory sentence that summarizes what 'zfs snapshot' does in the first place.  In that sentence, the text "different datasets" is intended to indicate that (again per the code) the same dataset cannot be specified multiple times on the command line.

### Types of changes

- [x] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md).
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

File: man/man8/zfs-snapshot.8 [hunk 1]
```diff
@@ -44,13 +44,19 @@
...
 .
 .Sh DESCRIPTION
-All previous modifications by successful system calls to the file system are
-part of the snapshots.
-Snapshots are taken atomically, so that all snapshots correspond to the same
-moment in time.
+Creates a snapshot of a dataset or multiple snapshots of different
+datasets.
+.Pp
+Snapshots are created atomically.  That is, a snapshot is a consistent
```
Comment (behlendorf) [inline man/man8/zfs-snapshot.8:50]: Thanks for the clarification and updated test.  Just a couple small mandoc formatting nits.

[new sentence, new line](https://man.openbsd.org/mandoc.1#new)
    (mdoc) A new sentence starts in the middle of a text line. Start it on a new input line to help formatters produce correct spacing.

File: man/man8/zfs-snapshot.8 [hunk 2]
```diff
@@ -44,13 +44,19 @@
...
+image of a dataset at a specific point in time; it includes all
+modifications to the dataset made by system calls that have
+successfully completed before that point in time.  Recursive snapshots
+created through the -r option are all created at the same time.
```
Comment (behlendorf) [inline man/man8/zfs-snapshot.8:54]: There's a formatting tag which should be added for the `-r` option.

```
created through the
.Fl r
option are all created at the same time.
```

Comment (shawnbayern) [reply man/man8/zfs-snapshot.8]: Thanks very much for the quick feedback  I've made and committed those changes.

File: man/man8/zfs-snapshot.8 (hunk 1 above)
Comment (shawnbayern) [reply man/man8/zfs-snapshot.8]: Thanks very much for the quick feedback. I've made and committed those changes.

**Review by behlendorf — APPROVED** [review]

**Comment by shawnbayern** [comment]
I noticed that the automated check flagged another minor formatting problem, so I've committed a fix.

(I'm a law professor and thought the rules for formatting legal citations were strict, but by comparison they're not!  :)  )

**Comment by behlendorf** [comment]
@shawnbayern thanks!

**Comment by shawnbayern** [comment]
Sorry to waste your time, but checkstyle now seems to be failing because my recent commit messages were over 72 characters per line.  (I knew to avoid that from the contributors' guide and avoided it in my initial writeup, but I think my last two commit messages were over that limit - sorry!)  I'm not sure of the cleanest way to fix that on GitHub to satisfy the checkstyle script; should I produce a new branch?  Thanks again for your help.

**Comment by behlendorf** [comment]
@shawnbayern no problem.  You'll want to squash your commits and then force up the PR.  This will kick of a new CI run.  Something like this:

```
git checkout patch-1
git rebase -i master
# Do the squash
git push --force shawnbayern patch-1
```

**Comment by shawnbayern** [comment]
@behlendorf thanks again!  I'll keep an eye out for any further problems.

**Pull request merged**